src/routes/search.rs

```diff
@@ -64,13 +86,53 @@ fn search_filter(manifest_filter: Option<ManifestFilter>) -> Result<Json<Vec<Man
         manifests.push(toml::from_str(&s)?);
     }
 
+    let mut sorter: fn(&Manifest, &Manifest) -> Ordering;
+    sorter = |a, b| b.metadata().created_at().cmp(&a.metadata().created_at()); // Default sort by create_at
+
     if let Some(filter) = manifest_filter {
         if let Some(name) = filter.name() {
             manifests.retain(|ref x: &Manifest| x.metadata().name().contains(name));
         }
         if let Some(category) = filter.category() {
             manifests.retain(|ref x: &Manifest| x.metadata().category().contains(category));
         }
+        if let Some(description) = filter.description() {
+            manifests.retain(|ref x: &Manifest| x.metadata().description().contains(description));
+        }
+        if let Some(tags) = filter.tags() {
+            manifests.retain(|ref x: &Manifest| x.metadata().tags().contains(tags));
+        }
+        if let Some(created_at) = filter.created_at() {
+            manifests.retain(|ref x: &Manifest| x.metadata().created_at().contains(created_at));
+        }
+        if let Some(order_by) = filter.order_by() {
+            match order_by.as_ref() {
+                "name_asc" => sorter = |a, b| a.metadata().name().cmp(&b.metadata().name()),
+                "name_desc" => sorter = |a, b| b.metadata().name().cmp(&a.metadata().name()),
+                "category_asc" => {
+                    sorter = |a, b| a.metadata().category().cmp(&b.metadata().category())
+                }
+                "category_desc" => {
+                    sorter = |a, b| b.metadata().category().cmp(&a.metadata().category())
+                }
+                "description_asc" => {
+                    sorter = |a, b| a.metadata().description().cmp(&b.metadata().description())
+                }
+                "description_desc" => {
+                    sorter = |a, b| b.metadata().description().cmp(&a.metadata().description())
+                }
+                "tags_asc" => sorter = |a, b| a.metadata().tags().cmp(&b.metadata().tags()),
+                "tags_desc" => sorter = |a, b| b.metadata().tags().cmp(&a.metadata().tags()),
+                "created_at_asc" => {
+                    sorter = |a, b| a.metadata().created_at().cmp(&b.metadata().created_at())
+                }
+                "created_at_desc" => {
+                    sorter = |a, b| b.metadata().created_at().cmp(&a.metadata().created_at())
+                }
+                _ => (),
+            }
+        }
     }
+    manifests.sort_by(sorter);
```

Comment:
You can probably simply sort the Vec when need instead of retaining the sorting algorithm to apply it at the end, na?